ext/sockets/sockets.c

```diff
@@ -1911,15 +1938,35 @@ PHP_FUNCTION(socket_set_option)
 				RETURN_THROWS();
 			}
 
-			convert_to_long(sec);
-			convert_to_long(usec);
+			zend_long valsec = zval_try_get_long(sec, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be of type int, %s given", sec_key, zend_zval_value_name(sec));
+				RETURN_THROWS();
+			}
+			zend_long valusec = zval_try_get_long(usec, &failed);
+			if (failed) {
+				zend_argument_type_error(4, "\"%s\" must be of type int, %s given", usec_key, zend_zval_value_name(usec));
+				RETURN_THROWS();
+			}
 #ifndef PHP_WIN32
-			tv.tv_sec = Z_LVAL_P(sec);
-			tv.tv_usec = Z_LVAL_P(usec);
+			tv.tv_sec = valsec;
+			tv.tv_usec = valusec;
 			optlen = sizeof(tv);
 			opt_ptr = &tv;
 #else
-			timeout = Z_LVAL_P(sec) * 1000 + Z_LVAL_P(usec) / 1000;
+			if (valsec < 0 || valsec > INT_MAX / 1000) {
+				zend_argument_value_error(4, "\"%s\" must be between 0 and %d", sec_key, INT_MAX / 1000);
+				RETURN_THROWS();
+			}
+
+			timeout = valsec * 1000;
+	
+			if (valusec < 0 || timeout > INT_MAX - (valusec / 1000)) {
+				zend_argument_value_error(4, "\"%s\" must be between 0 and %u", usec_key, (DWORD)(INT_MAX - (valusec / 1000)));
+				RETURN_THROWS();
+			}
```

Comment:
I don't think this works for very large `usec`. We may also need to take BC into account (currently, negative `usec` are likely allowed for Windows). Maybe leave the overflow checking for master; while technically UB can happen, I think wrap-around is happening in practice (and nothing really bad).